Lacework/lacework_module/lacework_connector.py

```diff
@@ -36,17 +37,34 @@ class LaceworkEventsTrigger(Connector):
 
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
+        self.from_date = ""
         self.context = PersistentJSON("context.json", self._data_path)
 
     @property
-    def cursor(self) -> str | None:
+    def most_recent_date_seen(self) -> datetime:
+        now = datetime.now(timezone.utc)
+
         with self.context as cache:
-            return str(cache.get("cursor"))
+            most_recent_date_seen_str = cache.get("most_recent_date_seen")
+
+            if most_recent_date_seen_str == None:
+                return now - timedelta(days=1)
 
-    @cursor.setter
-    def cursor(self, cursor: str) -> None:
+            most_recent_date_seen = isoparse(most_recent_date_seen_str)
+
+            one_week_ago = now - timedelta(days=7)
+            if most_recent_date_seen.replace(tzinfo=timezone.utc) < one_week_ago:
+                most_recent_date_seen = one_week_ago
+
+            return most_recent_date_seen
+
+    @most_recent_date_seen.setter
+    def most_recent_date_seen(self, recent_date: datetime) -> None:
+        # save the greatest date
+        if self.from_date == "" or isoparse(self.from_date) < recent_date:
+            self.from_date = str(recent_date)
```

Comment:
To represent a datetime as a string, it's preferable to use [`isoformat`](https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat):
```suggestion
            self.from_date = recent_date.isoformat()
```